### PR TITLE
Member status: Weakly up

### PR DESCRIPTION
**Pull request by Horusiath**

### CODE REVIEW

**Comment by Horusiath** [comment]
One note: if we're going to release v1.3.2, don't merge it yet. It's too serious to land in patch release, it's meant to for v1.4.

**Comment by Aaronontheweb** [comment]
1.4 aint happen for a while. Need a feature branch then.

**Comment by Horusiath** [comment]
Got some Hyperion serialization exceptions of MNTK for windows .NET core - something related to cluster sharding.

**Comment by Aaronontheweb** [comment]
Yeah, this is a breakage in the .NET Core Tooling - nothing to do with your PR.

**Comment by Horusiath** [comment]
I've turned this feature off by default - this way it won't affect current cluster protocol, and it still can be set manually. I think it could be merged directly.

**Comment by Aaronontheweb** [comment]
I need to check:

* [ ] How `WeaklyMemberUp` factors into leader election
* [ ] How `WeaklyMemberUp` impacts the bug fixes introduced in solving #2584

**Comment by Aaronontheweb** [comment]
Looks like we're in the clear on the node leader transition stuff, since you added `WeaklyUp` to the `AllowedTransitions` table, which is what our leader picking algorithm uses to determine what's the correct "next state" for each node in transition.

**Comment by Aaronontheweb** [comment]
Doesn't affect leader elections either:

```csharp
   private static readonly ImmutableHashSet<MemberStatus> LeaderMemberStatus =
            ImmutableHashSet.Create(MemberStatus.Up, MemberStatus.Leaving);
```

**Comment by Aaronontheweb** [comment]
Looks good to me. We'll get this in for v1.3.3

File: src/core/Akka.Cluster/ClusterDaemon.cs [hunk 1]
```diff
@@ -2138,6 +2142,39 @@ public void LeaderActions()
...
+
+            var changedMembers = localMembers
+                .Where(IsJoiningToWeaklyUp)
+                .Select(m => m.Copy(MemberStatus.WeaklyUp))
+                .ToImmutableSortedSet();
+
+            if (!changedMembers.IsEmpty)
+            {
+                // replace changed members
+                var newMembers = changedMembers.Union(localMembers);
```
Comment (Aaronontheweb) [inline src/core/Akka.Cluster/ClusterDaemon.cs]: Son of a bitch... I totally missed this when I originally reviewed the PR. It's going to great the same problems here that happened in https://github.com/akkadotnet/akka.net/issues/2584. This is a bug. I'll PR it per https://github.com/akkadotnet/akka.net/issues/3274

**Pull request merged**